### PR TITLE
Improve search results page

### PR DESCRIPTION
**Pull request by dasGoogle**
This is a PR to track some improvements to the search page. This includes proper sorting, a better content preview and maybe additional filters.

### CODE REVIEW

**Comment by dasGoogle** [comment]
Ping @jeriox @lukasrad02 for your suggestions

**Comment by jeriox** [comment]
I think you dropped the date (https://github.com/fsr-de/myHPI/pull/492), or is your example not a protocol? I think we should definetly include that.

Could you also provide a screenshot with multiple results? With the current layout I suspect that it could be hard to visually distinguish between different results as they are not as uniform as before

**Comment by dasGoogle** [comment]
image

Indeed, the screenshot was from a page with minutes content, the date remains in there for minutes

**Comment by jeriox** [comment]
I like it

**Comment by jeriox** [comment]
another suggestion from #345: restrict search to specific areas of the page (pages/minutes/even specific minutes lists or respective groups?) -> can also be a separate issue if you don't want to implement it in this PR

**Review by jeriox — APPROVED** [review]

File: myhpi/search/templatetags/search.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+from django.utils.safestring import mark_safe
+
+from myhpi.core.markdown.utils import render_markdown
+
+register = template.Library()
+
+
+@register.filter(name="highlight_query_markdown")
+# select 3 lines around first match and highlight query match
+def highlight_query(content, search_query, surrounding_lines=1):
```
Comment (frcroth) [inline myhpi/search/templatetags/search.py:14]: This function seems quite complex, maybe it could be tested?

**Pull request merged**